rootfs/etc/nginx/lua/metric_emitter.lua

```diff
@@ -0,0 +1,50 @@
+local socket = ngx.socket.udp
+local cjson = require('cjson')
+local assert = assert
+
+local _M = {
+  queue = {}
+}
+
+local function flush_queue()
+  local s = assert(socket())
+  assert(s:setpeername("127.0.0.1", 8000))
+  for _, v in ipairs(_M.queue) do
+    assert(s:send(v))
+  end
+  _M.queue = {}
+  assert(s:close())
+end
+
+function _M.call()
+  -- Create JSON Metrics Payload  --
+  local jsonPayload = cjson.encode({
+    host                 = ngx.var.host                             or "-",
+    status               = ngx.var.status                           or "-",
+    remoteAddr           = ngx.var.realip_remote_addr               or "-",
+    remoteUser           = ngx.var.remote_user                      or "-",
+    bytesSent            = tonumber(ngx.var.bytes_sent)             or -1,
+    protocol             = ngx.var.server_protocol                  or "-",
+    method               = ngx.var.request_method                   or "-",
+    path                 = ngx.var.uri                              or "-",
+    requestLength        = tonumber(ngx.var.request_length)         or -1,
+    requestDuration      = tonumber(ngx.var.request_time)           or -1,
+    upstreamName         = ngx.var.upstream                         or "-",
```

Comment:
`ngx.var.upstream` does not seem like a standard Nginx variable, are we setting it ourselves somewhere?

Reply:
Good catch, that was a mistake. It should've been `ngx.var.proxy_upstream_name`. Thinking about it, won't this be the same value all the time (upstream_balancer)?